jenkins/test-image/Dockerfile

```diff
@@ -43,6 +43,13 @@ RUN apt-get -o Acquire::Check-Valid-Until=false update && apt-get install -y \
 	--no-install-recommends \
 	&& rm -rf /var/lib/apt/lists/*
 
+# Download and symlink etcd. We need this for verification and integration tests.
+RUN export ETCD_VERSION=v3.0.4; \
```

Comment:
Modifying just the hack/install-etcd.sh script keeps the weirdness
localized. If we want to have etcd always available in $PATH and not use
hack/install-etcd.sh at all in our CI, that's fine too.

On Thu, Sep 8, 2016 at 9:01 PM, Erick Fejta <notifications@github.com>
wrote:

> In jenkins/test-image/Dockerfile
> <https://github.com/kubernetes/test-infra/pull/522#discussion_r78125897>:
>
> > @@ -43,6 +43,13 @@ RUN apt-get -o Acquire::Check-Valid-Until=false update && apt-get install -y \
> >  	--no-install-recommends \
> >  	&& rm -rf /var/lib/apt/lists/*
> >
> > +# Download and symlink etcd. We need this for verification and integration tests.
> > +RUN export ETCD_VERSION=v3.0.4; \
>
> This is super cool! I love that changing the ETCD version requires a) a PR
> to update this line and b) pushing a new image.
>
> On the other hand, is there any chance this can more closely follow the
> instructions at https://github.com/kubernetes/kubernetes/blob/master/docs/
> devel/testing.md#install-etcd-dependency? It is not clear that
> hack/install-etcd.sh does the following... in fact this says it installs it
> into ./third_party/etcd.
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/test-infra/pull/522/files/ab44d468ccc40a462a03620ca092a2511df5b6b8#r78125897>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAM7nEzdUXSyf0qOW1dkoS6Nrfm_y9IIks5qoNn8gaJpZM4J2WxO>
> .
>